src/components/Board.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react'
+import Square from './Square'
+import { checkWinner, checkStale } from '../utils'
+import Confetti from 'react-dom-confetti'
+import confettiConfig from './confetti-config'
+import Restart from './Restart'
+
+//component, destructured props of type BoardTypeProps
+const Board = ({
+  gameHistory,
+  addGameHistory,
+  currentTurn,
+}: BoardTypeProps) => {
+  const [gameState, setGameState] = useState<GameState>(gameHistory[0])
```

Comment:
by separating this out you have different sources of truth (gameHistory vs. gameState)

you have to update both when updating one, and having a mismatch between the two could be really bad.

I'd probably combine the two into a more complicated gameState object at the top level, and ideally use a [useReducer](https://reactjs.org/docs/hooks-reference.html#usereducer) to manage it. useReducer is really good for managing/manipulating complex state object